File: 1013-fibonacci-number/solution.c
int solve(int n,int* dp){
    if(n==0||n==1) return n;
    else if(dp[n] != -1) return dp[n];
    else return dp[n] = solve(n-1,dp) + solve(n-2,dp);
}
int fib(int n){
    int dp[n+1];
    memset(dp,-1,sizeof(dp));
    return solve(n,dp);
}

File: 1013-fibonacci-number/solution.java
class Solution {
    public int solve(int n,int[] arr){
        if(n==0 || n==1) return n;
        else if(arr[n] != -1) return arr[n];
        else return arr[n] = solve(n-1,arr) + solve(n-2,arr);
    }
    public int fib(int n) {
        int[] arr = new int[n+1];
        for(int i = 0;i<n+1;i++){
            arr[i] = -1;
        }
        return solve(n,arr);
    }
}
